importers/algod/algod_importer.go

```diff
@@ -36,31 +38,66 @@ func New() importers.Importer {
 	return &algodImporter{}
 }
 
+// Constructor is the Constructor implementation for the "algod" importer
+type Constructor struct{}
+
+// New initializes a blockProcessorConstructor
+func (c *Constructor) New() importers.Importer {
+	return &algodImporter{}
+}
+
 func (algodImp *algodImporter) Metadata() importers.ImporterMetadata {
 	return algodImporterMetadata
 }
 
-func (algodImp *algodImporter) Init(ctx context.Context, cfg plugins.PluginConfig, logger *logrus.Logger) error {
+// package-wide init function
+func init() {
+	importers.RegisterImporter(importerName, &Constructor{})
+}
+
+func (algodImp *algodImporter) Init(ctx context.Context, cfg plugins.PluginConfig, logger *logrus.Logger) (*bookkeeping.Genesis, error) {
 	algodImp.ctx, algodImp.cancel = context.WithCancel(ctx)
 	algodImp.logger = logger
 	if err := algodImp.unmarhshalConfig(string(cfg)); err != nil {
-		return fmt.Errorf("connect failure in unmarshalConfig: %v", err)
+		return nil, fmt.Errorf("connect failure in unmarshalConfig: %v", err)
 	}
 	var client *algod.Client
 	u, err := url.Parse(algodImp.cfg.NetAddr)
 	if err != nil {
-		return err
+		return nil, err
 	}
+
+	netAddrIsEmpty := algodImp.cfg.NetAddr == ""
+
 	if u.Scheme != "http" && u.Scheme != "https" {
 		algodImp.cfg.NetAddr = "http://" + algodImp.cfg.NetAddr
 		algodImp.logger.Infof("Algod Importer added http prefix to NetAddr: %s", algodImp.cfg.NetAddr)
 	}
 	client, err = algod.MakeClient(algodImp.cfg.NetAddr, algodImp.cfg.Token)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	algodImp.aclient = client
-	return err
+
+	// For testing check against nil
+	if !netAddrIsEmpty {
```

Comment:
Are we sure we want to return a `nil` error when the algod address is empty?

Reply:
How would we indicate that we don't want to retrieve a genesis file for testing purposes?

Reply:
Use the mock algod server for the init test in the same way that we do for the get block test.

It's not a huge issue since an empty address in your algod client will fail pretty quickly when trying to retrieve a block, but we can detect here that it's not going to work if it has no address so it would maybe be easier to debug if we caught it here.

Reply:
+1